### PR TITLE
adds the ability to tear and burn flags

### PR DESCRIPTION
**Pull request by dragoslava**
Adds the ability to tear flags down and burn them

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/obj/contraband.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
flag_torn||flag_torn|Created
</details>

File: code/game/objects/structures/signs.dm [hunk 1]
```diff
@@ -553,3 +553,35 @@
...
+
+		if(!do_after(user, 2 SECONDS, act_target = src))
+			return 0
+
+
+		visible_message("<span class='warning'>\The [user] rips \the flag in a single, decisive motion!</span>" )
+		playsound(src.loc, 'sound/items/poster_ripped.ogg', 100, 1)
+		icon_state = "poster_ripped"
+		name = "ripped poster"
+		desc = "You can't make out anything from the flags's original print. It's ruined."
```
Comment (Lohikar) [inline code/game/objects/structures/signs.dm]: `flags's` is invalid. Use `flag's`, there's only one flag here.

File: code/game/objects/structures/signs.dm [hunk 2]
```diff
@@ -553,3 +553,35 @@
...
+		add_fingerprint(user)
+
+/obj/structure/sign/flag/attackby(obj/item/weapon/W as obj, mob/user as mob)
+    ..()
+
+    if(istype(W, /obj/item/weapon/flame/lighter))
+
+        visible_message("<span class='warning'>\The [user] starts to burn \the flag down!</span>")
```
Comment (Lohikar) [inline code/game/objects/structures/signs.dm]: Use of text macros on a literal (`\the flag`) is pointless, and the down here is redundant.

File: code/game/objects/structures/signs.dm [hunk 3]
```diff
@@ -553,3 +553,35 @@
...
+
+        if(!do_after(user, 2 SECONDS, act_target = src))
+            return 0
+        visible_message("<span class='warning'>\The [user] burns \the flag down. Viva la Revolucion!</span>")
```
Comment (Lohikar) [inline code/game/objects/structures/signs.dm]: Vis messages should be objective. This is subjective.

File: code/game/objects/structures/signs.dm [hunk 4]
```diff
@@ -553,3 +553,35 @@
 	P.desc = desc
 	qdel(src)
 
+
+/obj/structure/sign/flag/attack_hand(mob/user as mob)
+
+
+	if(alert("Do I want to rip the flag from the wall?","You think...","Yes","No") == "Yes")
```
Comment (Lohikar) [inline code/game/objects/structures/signs.dm]: Why not put the flag's name in this message? Prob. should be in 2nd person too.

File: code/game/objects/structures/signs.dm [hunk 5]
```diff
@@ -553,3 +553,35 @@
...
+/obj/structure/sign/flag/attack_hand(mob/user as mob)
+
+
+	if(alert("Do I want to rip the flag from the wall?","You think...","Yes","No") == "Yes")
+
+		if(!do_after(user, 2 SECONDS, act_target = src))
+			return 0
+
+
+		visible_message("<span class='warning'>\The [user] rips \the flag in a single, decisive motion!</span>" )
```
Comment (skull132) [inline code/game/objects/structures/signs.dm]: Replace `\the flag` with `\the [src]`.

File: code/game/objects/structures/signs.dm (hunk 2 above)
Comment (skull132) [reply code/game/objects/structures/signs.dm]: Again, use `\the [src]` instead of `\the flag`.

File: code/game/objects/structures/signs.dm [hunk 6]
```diff
@@ -553,3 +553,35 @@
...
+		playsound(src.loc, 'sound/items/poster_ripped.ogg', 100, 1)
+		icon_state = "poster_ripped"
+		name = "ripped poster"
+		desc = "You can't make out anything from the flags's original print. It's ruined."
+		add_fingerprint(user)
+
+/obj/structure/sign/flag/attackby(obj/item/weapon/W as obj, mob/user as mob)
+    ..()
```
Comment (skull132) [inline code/game/objects/structures/signs.dm]: Starting from here, the file is indented with spaces. Please reindent with tabs.

File: code/game/objects/structures/signs.dm [hunk 7]
```diff
@@ -553,3 +553,35 @@
...
+
+        visible_message("<span class='warning'>\The [user] starts to burn \the flag down!</span>")
+
+        if(!do_after(user, 2 SECONDS, act_target = src))
+            return 0
+        visible_message("<span class='warning'>\The [user] burns \the flag down. Viva la Revolucion!</span>")
+        playsound(src.loc, 'sound/items/zippo_on.ogg', 100, 1)
+        new /obj/effect/decal/cleanable/ash(src.loc)
+
+    qdel(src)
```
Comment (skull132) [inline code/game/objects/structures/signs.dm]: This needs to be intented one more level. Right now, any call to `attackby` will delete the flag.

**Pull request merged**